### PR TITLE
Feature/321 reformat teams

### PR DESCRIPTION
**Pull request by kcinay055679**

### CODE REVIEW

File: backend/src/main/java/ch/puzzle/okr/service/validation/TeamValidationService.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+@Service
+public class TeamValidationService extends ValidationBase<Team, Long> {
+
+    public TeamValidationService(TeamPersistenceService teamPersistenceService) {
+        super(teamPersistenceService);
+    }
+
+    @Override
+    public void validateOnGet(Long id) {
+        isIdNull(id);
```
Comment (MarcEgliP) [inline backend/src/main/java/ch/puzzle/okr/service/validation/TeamValidationService.java:18]: Discussed changes

File: backend/src/test/java/ch/puzzle/okr/service/TeamServiceTest.java [hunk 2]
```diff
@@ -64,179 +61,69 @@ private static Stream<Arguments> shouldGetTeamsByIds() {
...
-    @ParameterizedTest
-    @MethodSource
-    void shouldGetTeamsByIds(List<Long> teamIds, List<Team> teamList, int expectedTeamsAmount) {
-        Mockito.when(teamRepository.findByName(Constants.TEAM_PUZZLE)).thenReturn(Optional.of(teamPuzzle));
-        Mockito.when(teamRepository.findAllByIdInAndNameNotOrderByNameAsc(teamIds, Constants.TEAM_PUZZLE))
-                .thenReturn(teamList);
-        assertEquals(expectedTeamsAmount, teamService.getAllTeams(teamIds).size());
-    }
-
+    //
```
Comment (MarcEgliP) [inline backend/src/test/java/ch/puzzle/okr/service/TeamServiceTest.java:71]: Delete unused line

File: backend/src/test/java/ch/puzzle/okr/service/validation/TeamValidationServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        teamWithIdNull = Team.Builder.builder().withId(null).withName("Team null").build();
+
+        when(teamPersistenceService.findById(1L)).thenReturn(team1);
+        when(teamPersistenceService.getModelName()).thenReturn("Team");
+        doThrow(new ResponseStatusException(HttpStatus.NOT_FOUND,
+                String.format("%s with id %s not found", teamPersistenceService.getModelName(), 2L)))
+                        .when(teamPersistenceService).findById(2L);
+    }
+
+    @Spy
```
Comment (MarcEgliP) [inline backend/src/test/java/ch/puzzle/okr/service/validation/TeamValidationServiceTest.java:50]: Why Spy?

**Review by MarcEgliP — CHANGES_REQUESTED** [review]
Looks good so far, implement the requested changes please!

File: backend/src/test/java/ch/puzzle/okr/service/validation/TeamValidationServiceTest.java (hunk 3 above)
Comment (kcinay055679) [reply backend/src/test/java/ch/puzzle/okr/service/validation/TeamValidationServiceTest.java:50]: bro i genuinely don't know

Comment (kcinay055679) [reply backend/src/test/java/ch/puzzle/okr/service/validation/TeamValidationServiceTest.java:50]: nvm this is like this so i can oberserve and verify function calls

File: backend/src/main/java/ch/puzzle/okr/controller/v2/TeamControllerV2.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Operation(summary = "Get Teams", description = "Get all Teams from db as well as all active objectives from chosen quarter")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Returned all Teams with active objective in quarter", content = {
+                    @Content(mediaType = "application/json", schema = @Schema(implementation = TeamDto.class)) }), })
+    @GetMapping("/{id}")
+    public List<TeamDto> getAllTeams(
+            @Parameter(description = "The ID of a Team to get a list of its Objectives.", required = false) @PathVariable long id) {
+        this.registerNewUserService.registerNewUser(SecurityContextHolder.getContext());
```
Comment (peggimann) [inline backend/src/main/java/ch/puzzle/okr/controller/v2/TeamControllerV2.java:42]: rename to quarterID

File: backend/src/main/java/ch/puzzle/okr/dto/TeamDto.java [hunk 5]
```diff
@@ -1,4 +1,7 @@
 package ch.puzzle.okr.dto;
 
-public record TeamDto(Long id, String name) {
+public record TeamDto(Long id, String name, Integer activeObjectives) {
+    public TeamDto(Long id, String name) {
```
Comment (peggimann) [inline backend/src/main/java/ch/puzzle/okr/dto/TeamDto.java:4]: add @Deprecated

File: backend/src/main/java/ch/puzzle/okr/mapper/TeamMapper.java [hunk 6]
```diff
@@ -2,10 +2,28 @@
...
+        this.quarterService = quarterService;
+    }
+
+    public TeamDto toDto(Team team, Long quarterId) {
+        long chosenQuarterId = quarterId == null ? quarterService.getActiveQuarter().getId() : quarterId;
+        Integer activeObjectives = teamService.activeObjectivesAmountOfTeam(team, chosenQuarterId);
+        return new TeamDto(team.getId(), team.getName(), activeObjectives);
+    }
+
     public TeamDto toDto(Team team) {
```
Comment (peggimann) [inline backend/src/main/java/ch/puzzle/okr/mapper/TeamMapper.java:27]: add @Deprecated

File: backend/src/main/java/ch/puzzle/okr/service/ObjectiveService.java [hunk 7]
```diff
@@ -2,10 +2,11 @@
 
 import ch.puzzle.okr.models.KeyResult;
 import ch.puzzle.okr.models.Objective;
+import ch.puzzle.okr.models.Quarter;
+import ch.puzzle.okr.models.Team;
 import ch.puzzle.okr.repository.KeyResultRepository;
-import ch.puzzle.okr.repository.MeasureRepository;
 import ch.puzzle.okr.repository.ObjectiveRepository;
-import ch.puzzle.okr.repository.TeamRepository;
+import ch.puzzle.okr.service.validation.TeamValidationService;
```
Comment (peggimann) [inline backend/src/main/java/ch/puzzle/okr/service/ObjectiveService.java:9]: remove unused validationservice

File: backend/src/test/java/ch/puzzle/okr/service/ObjectiveServiceTest.java [hunk 8]
```diff
@@ -5,6 +5,7 @@
 import ch.puzzle.okr.repository.MeasureRepository;
 import ch.puzzle.okr.repository.ObjectiveRepository;
 import ch.puzzle.okr.repository.TeamRepository;
+import ch.puzzle.okr.service.validation.TeamValidationService;
```
Comment (peggimann) [inline backend/src/test/java/ch/puzzle/okr/service/ObjectiveServiceTest.java:8]: can be removed

**Pull request merged**